src/compiler/diagnostic_information_map_generated.rs

```diff
@@ -0,0 +1,27 @@
+use crate::{DiagnosticCategory, DiagnosticMessage};
+
+const fn diag(
+    code: u32,
+    category: DiagnosticCategory,
+    key: &'static str,
+    message: &'static str,
+) -> DiagnosticMessage {
+    DiagnosticMessage {
+        code,
+        category,
+        key,
+        message,
+    }
+}
+
+#[non_exhaustive]
+pub struct Diagnostics;
+#[allow(non_upper_case_globals)]
+impl Diagnostics {
+    pub const _0_expected: DiagnosticMessage = diag(
+        1005,
+        DiagnosticCategory::Error,
+        "_0_expected_1005",
+        "'{0}' expected.",
+    );
+}
```

Comment:
Similar to `Debug`, I'm not sure why this is a struct rather than just functions.